src/androidTest/java/com/couchbase/lite/DatabaseTest.java

```diff
@@ -335,4 +338,73 @@ public void testAndroid2MLimit() throws Exception {
         assertNotNull(properties);
         assertEquals(content, properties.get("content"));
     }
+
+    // Database_Tests.m : test18_Attachments
+    public void testAttachments() throws Exception {
+        Map<String, Object> properties = new HashMap<String, Object>();
+        properties.put("testName", "testAttachments");
+        properties.put("count", 1);
+
+        final Document doc = createDocumentWithProperties(database, properties);
+        SavedRevision rev = doc.getCurrentRevision();
+        assertEquals(0, rev.getAttachments().size());
+        assertEquals(0, rev.getAttachmentNames().size());
+        assertNull(rev.getAttachment("index.html"));
+
+        String content = "This is a test attachments!";
+        ByteArrayInputStream body = new ByteArrayInputStream(content.getBytes());
+        UnsavedRevision rev2 = doc.createRevision();
+        rev2.setAttachment("index.html", "text/plain; charset=utf-8", body);
+
+        assertEquals(1, rev2.getAttachments().size());
+        assertEquals(1, rev2.getAttachmentNames().size());
+        assertEquals("index.html", rev2.getAttachmentNames().get(0));
+        Attachment attach = rev2.getAttachment("index.html");
+        assertNotNull(attach);
+        assertNull(attach.getRevision()); // No revision set
+        assertNull(attach.getDocument()); // No revision set
+        assertEquals("index.html", attach.getName());
+        assertEquals("text/plain; charset=utf-8", attach.getContentType());
+
+        SavedRevision rev3 = rev2.save();
+        assertNotNull(rev3);
+        assertEquals(1, rev3.getAttachments().size());
+        assertEquals(1, rev3.getAttachmentNames().size());
+        assertEquals("index.html", rev3.getAttachmentNames().get(0));
+
+        attach = rev3.getAttachment("index.html");
+        assertNotNull(attach);
+        assertNotNull(attach.getRevision());
+        assertNotNull(attach.getDocument());
+        assertEquals(doc, attach.getDocument());
+        assertEquals("index.html", attach.getName());
+        assertEquals("text/plain; charset=utf-8", attach.getContentType());
+        assertTrue(Arrays.equals(content.getBytes(), TextUtils.read(attach.getContent())));
+
+        // Look at the attachment's file:
+        URL bodyURL = attach.getContentURL();
+        if (isEncryptedAttachmentStore()) {
+            assertNull(bodyURL);
```

Comment:
I believe the code (method) level documentation is done. Here is the ticket for API documentation level. 

https://github.com/couchbaselabs/couchbase-lite-api/issues/130